docs/cohere.md

```diff
@@ -0,0 +1,445 @@
+   embed-english-v2.0            | {embed,classify} | f         |            512 |                                                                                            | {}
+   embed-multilingual-light-v3.0 | {embed,classify} | f         |            512 | https://storage.googleapis.com/cohere-public/tokenizers/embed-multilingual-light-v3.0.json | {}
+   embed-multilingual-v3.0       | {embed,classify} | f         |            512 | https://storage.googleapis.com/cohere-public/tokenizers/embed-multilingual-v3.0.json       | {}
+   embed-multilingual-v2.0       | {embed,classify} | f         |            256 | https://storage.googleapis.com/cohere-public/tokenizers/embed-multilingual-v2.0.json       | {}
+   embed-english-v3.0            | {embed,classify} | f         |            512 | https://storage.googleapis.com/cohere-public/tokenizers/embed-english-v3.0.json            | {}
+   embed-english-light-v3.0      | {embed,classify} | f         |            512 | https://storage.googleapis.com/cohere-public/tokenizers/embed-english-light-v3.0.json      | {}
+  (7 rows)
+  ```
+
+* List the default model for a given endpoint.
+  
+  ```sql
+  select * 
+  from cohere_list_models(_endpoint=>'generate', _default_only=>true);
+  ```
+  
+  Results
+  
+  ```text
+    name   |         endpoints         | finetuned | context_length |                            tokenizer_url                             | default_endpoints 
+  ---------+---------------------------+-----------+----------------+----------------------------------------------------------------------+-------------------
+   command | {generate,summarize,chat} | f         |           4096 | https://storage.googleapis.com/cohere-public/tokenizers/command.json | {generate}
+  (1 row)
+  ```
+
+### cohere_tokenize
+
+Tokenize text content.
+
+```sql
+select cohere_tokenize
+( 'command'
+, 'One of the best programming skills you can have is knowing when to walk away for awhile.'
+);
+```
+
+Results:
+
+```text
+                                      cohere_tokenize                                       
+--------------------------------------------------------------------------------------------
+ {5256,1707,1682,2383,9461,4696,1739,1863,1871,1740,9397,2112,1705,4066,3465,1742,38700,21}
+(1 row)
+```
+
+### cohere_detokenize
+
+Reverse the tokenize process.
+
+```sql
+select cohere_detokenize
+( 'command'
+, array[14485,38374,2630,2060,2252,5164,4905,21,2744,2628,1675,3094,23407,21]
+);
+```
+
+Results:
+
+```text
+                              cohere_detokenize                               
+------------------------------------------------------------------------------
+ Good programmers don't just write programs. They build a working vocabulary.
+(1 row)
+```
+
+### cohere_embed
+
+Embed content.
+
+```sql
+select cohere_embed
+( 'embed-english-light-v3.0'
+, 'if a woodchuck could chuck wood, a woodchuck would chuck as much wood as he could'
+, _input_type=>'search_document'
```

Comment:
Just thinking loud, we may add the `_input_type` with a default just to allow us to have a common method signature between the models.

```
Caused by PG::ExternalRoutineException: ERROR:  cohere.errors.bad_request_error.BadRequestError: status_code: 400, body: {'message': 'invalid request: valid input_type must be provided with the provided model'}
```